File: 1666-MakeTheStringGreat/1666-MakeTheStringGreat.java
// Last updated: 10/8/2025, 9:48:54 PM
class Solution {
    public String makeGood(String s) {
        Stack<Character> stack=new Stack<>();
        for(char c:s.toCharArray()){
            if (!stack.isEmpty()&& Math.abs(stack.peek()-c)==32){
                    stack.pop();
        }else{stack.push(c);}
        }
        StringBuilder st=new StringBuilder();
        for(char item :stack){
            st.append(item);
        }
        return st.toString();
    }
}

File: 1666-MakeTheStringGreat/1666-MakeTheStringGreat.py
# Last updated: 10/8/2025, 9:48:52 PM
class Solution:
    def makeGood(self, s: str) -> str:
        stack=[]
        
        for c in s:
            
            #"leEeetcode"
            if stack and stack[-1]!=c and stack[-1].upper()==c.upper():
                    stack.pop()
                    continue
                
            stack.append(c)
        return "".join(stack)
        